contracts/src/7683/T1ERC7683.sol

```diff
@@ -170,20 +105,191 @@ contract T1ERC7683 is Base7683, OwnableUpgradeable, PausableUpgradeable {
+    /// @dev Intended to improve standardized integration of various order types and settlement contracts
+    /// @param _order The OnchainCrossChainOrder definition
+    /// @return _resolvedOrder ResolvedCrossChainOrder hydrated order data including the inputs and outputs of the order
+    function resolve(OnchainCrossChainOrder calldata _order)
+        public
+        view
+        virtual
+        returns (ResolvedCrossChainOrder memory _resolvedOrder)
+    {
+        (_resolvedOrder,,) = _resolveOrder(_order);
+    }
+
+    /// @dev Resolves a GaslessCrossChainOrder.
+    /// @param _order The GaslessCrossChainOrder to resolve.
+    /// NOT USED _originFillerData Any filler-defined data required by the settler
+    /// @return A ResolvedCrossChainOrder structure.
+    /// @return The order ID.
+    /// @return The order nonce.
+    function _resolveOrder(
+        GaslessCrossChainOrder memory _order,
+        bytes calldata
+    )
+        internal
+        view
+        returns (ResolvedCrossChainOrder memory, bytes32, uint256)
+    {
+        return _resolvedOrder(
+            _order.orderDataType, _order.user, _order.openDeadline, _order.fillDeadline, _order.orderData
+        );
+    }
+
+    /// @notice Resolves a OnchainCrossChainOrder.
+    /// @param _order The OnchainCrossChainOrder to resolve.
+    /// @return A ResolvedCrossChainOrder structure.
+    /// @return The order ID.
+    /// @return The order nonce.
+    function _resolveOrder(OnchainCrossChainOrder memory _order)
+        internal
+        view
+        returns (ResolvedCrossChainOrder memory, bytes32, uint256)
+    {
+        return _resolvedOrder(_order.orderDataType, msg.sender, type(uint32).max, _order.fillDeadline, _order.orderData);
+    }
+
+    /// @dev Resolves an order into a ResolvedCrossChainOrder structure.
+    /// @param _orderType The type of the order.
+    /// @param _sender The sender of the order.
+    /// @param _openDeadline The open deadline of the order.
+    /// @param _fillDeadline The fill deadline of the order.
+    /// @param _orderData The data of the order.
+    /// @return resolvedOrder A ResolvedCrossChainOrder structure.
+    /// @return orderId The order ID.
+    /// @return nonce The order nonce.
+    function _resolvedOrder(
+        bytes32 _orderType,
+        address _sender,
+        uint32 _openDeadline,
+        uint32 _fillDeadline,
+        bytes memory _orderData
+    )
+        internal
+        view
+        returns (ResolvedCrossChainOrder memory resolvedOrder, bytes32 orderId, uint256 nonce)
+    {
+        if (_orderType != OrderEncoder.orderDataType()) revert InvalidOrderType(_orderType);
+
+        // IDEA: _orderData should not be directly typed as OrderData, it should contain information that is not
+        // present on the type used for open the order. So _fillDeadline and _user should be passed as arguments
+        OrderData memory orderData = OrderEncoder.decode(_orderData);
```

Comment:
i'm not following this. could you please explain?

Reply:
Yes it is horribly worded, not sure why I was so unclear.
What I meant is that there are values in OrderData that are already present in the global Order struct like `user` and `sender`, `fillDeadline`, `originChainId` and `originDomain`.
So we don't need to have them in the OrderData struct and should remove them. But I left them for now as our offchain infra might use them.
I will improve the comment.

Reply:
Got it, thanks